article/_posts/2020-08-13-DisentanglingByFactorising.md

```diff
@@ -0,0 +1,105 @@
+method (InfoWGAN-GP)
+
+
+# Introduction
+The authors analyze the disentanglement and reconstruction aspects of $$\beta$$-VAE, which essentially allows to weight
+the KL divergence term in the classic VAE loss. The $$\beta$$-VAE objective can be formulated like follows, where $$\beta \geq 1$$:
+
+$$
+\frac{1}{N} \sum_{i=1}^{N} [\mathbb{E}_{q(z|x^{(i)})}[\log p(x^{(i)}|z)] - \beta KL(q(z|x^{(i)})~\|~p(z))]
+$$
+
+The first term is the reconstruction error, and the second term is the complexity penalty. This second term can be
+reformulated like below, given $$q(z)$$ the distribution of representations for the entire data set and $$I(x;z)$$ the
+mutual information between an input $$x$$ and its encoding $$z$$ under the joint distribution $$p_{data}(x)q(z|x)$$.
+
+$$
+\mathbb{E}_{p_{data}(x)}[KL(q(z|x)~\|~p(z))] = I(x;z) + KL(q(z)~\|~p(z))
+$$
+
+What this equivalent formulation highlights is that while penalizing the original KL divergence with $$\beta \geq 1$$
+term pushes $$q(z)$$ towards $$p(z)$$ (encouraging disentanglement), it also reduces the information about $$x$$ stored
+in $$z$$, and therefore the reconstruction accuracy. The authors state that penalizing $$I(x;z)$$ "might be neither
+necessary nor desirable for disentangling".
+
+
+# Methods
+
+## FactorVAE
+Given the claim about the mutual information term, the authors propose to motivate FactorVAE with an objective that
+directly encourages the independence in the code distribution:
+
+$$
+\frac{1}{N} \sum_{i=1}^{N} [\mathbb{E}_{q(z|x^{(i)})}[\log p(x^{(i)}|z)] - KL(q(z|x^{(i)})~\|~p(z))] 
+- \gamma KL(q(z)~\|~\bar{q}(z))
+$$
+
+where $$\bar{q}(z) = \prod_{j=1}^{d} q(z_{j})$$. This is also a lower bound on the marginal log likelihood. $$KL(q(z)~\|~\bar{q}(z))$$
+is also known as *Total Correlation* (TC), and is intractable in this case, since it would require passing through the
+entire dataset for each evaluation.
+
+To circumvent this issue, the authors explain that it is possible to sample from both $$q(z)$$ and $$\bar{q}(z)$$
+(details on how to this are discussed in the paper), which then allows them to use the *density-ratio trick* to minimize
+the KL divergence. The *density-ratio trick* consists of training a discriminator $$D$$ (in their case an MLP) to
+estimate wether a sample comes from $$q(z)$$ or $$\bar{q}(z)$$. This allows them to rephrase the TC as:
+
+$$
+TC(z) = KL(q(z)~\|~\bar{q}(z)) = \mathbb{E}_{q(z)}[\log \frac{q(Z)}{\bar{q}(z)}] \approx \mathbb{E}_{q(z)}[\log \frac{D(z)}{1 - D(z)}]
+$$
+
+The discriminator and VAE are trained jointly, and the training algorithm for FactorVAE is given below:
+![](/article/images/DisentanglingByFactorising/algorithm2.jpg)
+
+As for the InfoWGAN-GP, it builds on top of InfoGAN, which is an alternative to $$\beta$$-VAE that learns disentangled
+representations. Little empirical comparisons exist between VAE-based methods and InfoGAN (due to GAN training stability).
```

Comment:
_due to GAN training stability_ : it's rather the instability or stability issues, right?

Reply:
Addressed in 2df1da9.